File: app/src/main/java/com/abelcht/n01f2smwc/smartwatch/communication/packages/ChangeDateTimePackage.kt
package com.abelcht.n01f2smwc.smartwatch.communication.packages;

class ChangeDateTimePackage(
    hours: Int, minutes: Int, seconds: Int, day: Int, month: Int, year: Int
) : SmartWatchPackage(
    0x01.toByte(),
    arrayOf(
        (year - 2000).toByte(), month.toByte(), day.toByte(), hours.toByte(), minutes.toByte(),
        seconds.toByte()
    )
)

File: python_framework/packages/ChangeDateTimePackage.py
from datetime import datetime

from packages.SmartWatchPackage import SmartWatchPackage


class ChangeDateTimePackage(SmartWatchPackage):
    def __init__(self, date: datetime):
        # Package data
        hour: int = date.hour
        minute: int = date.minute
        second: int = date.second
        day: int = date.day
        month: int = date.month
        year: int = date.year - 2000

        message_type = 0x01
        content = [year, month, day, hour, minute, second]
        super().__init__(message_type, content)
